woocommerce/woocommerce
# [Accessibility] Associate email field and payment option radio with their descriptions on Checkout

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #51793 .
Closes #51794 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

**Testing the Email address input**
1. Make sure your Checkout page uses the block-based version.
2. Got to WooCommerce > Settings > Accounts & Privacy and enable the guest checkout option.
3. Make sure you are not logged in the store.
4. Add any product to the cart and go to the Checkout page.
5. Turn on the screen reader.
6. Focus on the "Email address" field and verify the "You are currently checking out as a guest." description is associated with the input.
![Screenshot 2024-11-19 at 11 36 47](https://github.com/user-attachments/assets/ab7d0592-9a59-4811-a2c6-5d28f6cd3d07)
7. Verify the "Email address" input has the `aria-describedby` attribute whose value is the same as the `id` attribute of its description.

**Testing the Payment option radios**
1. Add at least two payment options to your store.
2. After adding a product to the Cart, go to the Checkout page.
3. Scroll down to the "Payment options" section.
4. Focus on the options and verify the description is associated with the input.
![Screenshot 2024-11-19 at 11 44 38](https://github.com/user-attachments/assets/4e5120f1-3fab-47ea-8edd-d25aaf5e94e8)
5. Verify the payment radio input has the `aria-describedby` attribute whose value is the same as the `id` attribute of its
description.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
